polyfactory/factories/base.py

```diff
@@ -319,7 +320,10 @@ def _handle_factory_field(
         if isinstance(field_value, Fixture):
             return field_value.to_value()
 
-        return field_value() if callable(field_value) else field_value
+        if callable(field_value):
+            return field_value()
+
+        return field_value if isinstance(field_value, Hashable) else copy.deepcopy(field_value)
```

Comment:
What should happen if the user has defined a `__hash__` method (the instance check ends up being `True`)?

Should we consider changing this into a warning instead of doing this deepcopy?

Reply:
It will be copied in that case. This is copying the behaviour of pydantic (see https://docs.pydantic.dev/latest/concepts/models/#fields-with-non-hashable-default-values). I can add documentation for this behaviour?

Alternative approach, would be to warn or maybe raise here like dataclasses.

Reply:
Cool. Sounds good to me.